[2022-02-27 15:56:20,650] {taskinstance.py:1032} INFO - Dependencies all met for <TaskInstance: Download_Stock_Price.save_to_database manual__2022-02-27T07:56:13.424155+00:00 [queued]>
[2022-02-27 15:56:20,667] {taskinstance.py:1032} INFO - Dependencies all met for <TaskInstance: Download_Stock_Price.save_to_database manual__2022-02-27T07:56:13.424155+00:00 [queued]>
[2022-02-27 15:56:20,667] {taskinstance.py:1238} INFO - 
--------------------------------------------------------------------------------
[2022-02-27 15:56:20,667] {taskinstance.py:1239} INFO - Starting attempt 1 of 1
[2022-02-27 15:56:20,667] {taskinstance.py:1240} INFO - 
--------------------------------------------------------------------------------
[2022-02-27 15:56:20,685] {taskinstance.py:1259} INFO - Executing <Task(PythonOperator): save_to_database> on 2022-02-27 07:56:13.424155+00:00
[2022-02-27 15:56:20,689] {standard_task_runner.py:52} INFO - Started process 49583 to run task
[2022-02-27 15:56:20,694] {standard_task_runner.py:76} INFO - Running: ['***', 'tasks', 'run', 'Download_Stock_Price', 'save_to_database', 'manual__2022-02-27T07:56:13.424155+00:00', '--job-id', '15', '--raw', '--subdir', 'DAGS_FOLDER/download_stock_price.py', '--cfg-path', '/var/folders/7y/kt4ctk651rggrj3r9z9t02wc0000gp/T/tmpie3tscup', '--error-file', '/var/folders/7y/kt4ctk651rggrj3r9z9t02wc0000gp/T/tmpnk1qtms3']
[2022-02-27 15:56:20,696] {standard_task_runner.py:77} INFO - Job 15: Subtask save_to_database
[2022-02-27 15:56:20,824] {logging_mixin.py:109} INFO - Running <TaskInstance: Download_Stock_Price.save_to_database manual__2022-02-27T07:56:13.424155+00:00 [running]> on host 1.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.ip6.arpa
[2022-02-27 15:56:20,914] {taskinstance.py:1424} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=andyhsu
AIRFLOW_CTX_DAG_ID=Download_Stock_Price
AIRFLOW_CTX_TASK_ID=save_to_database
AIRFLOW_CTX_EXECUTION_DATE=2022-02-27T07:56:13.424155+00:00
AIRFLOW_CTX_DAG_RUN_ID=manual__2022-02-27T07:56:13.424155+00:00
[2022-02-27 15:56:20,918] {logging_mixin.py:109} WARNING - /Users/andyhsu/tutorial-env/lib/python3.9/site-packages/***/utils/context.py:152 AirflowContextDeprecationWarning: Accessing 'execution_date' from the template is deprecated and will be removed in a future version. Please use 'data_interval_start' or 'logical_date' instead.
[2022-02-27 15:56:20,919] {logging_mixin.py:109} WARNING - /Users/andyhsu/tutorial-env/lib/python3.9/site-packages/***/utils/context.py:152 AirflowContextDeprecationWarning: Accessing 'next_ds' from the template is deprecated and will be removed in a future version. Please use '{{ data_interval_end | ds }}' instead.
[2022-02-27 15:56:20,919] {logging_mixin.py:109} WARNING - /Users/andyhsu/tutorial-env/lib/python3.9/site-packages/***/utils/context.py:152 AirflowContextDeprecationWarning: Accessing 'next_ds_nodash' from the template is deprecated and will be removed in a future version. Please use '{{ data_interval_end | ds_nodash }}' instead.
[2022-02-27 15:56:20,920] {logging_mixin.py:109} WARNING - /Users/andyhsu/tutorial-env/lib/python3.9/site-packages/***/utils/context.py:152 AirflowContextDeprecationWarning: Accessing 'next_execution_date' from the template is deprecated and will be removed in a future version. Please use 'data_interval_end' instead.
[2022-02-27 15:56:20,921] {logging_mixin.py:109} WARNING - /Users/andyhsu/tutorial-env/lib/python3.9/site-packages/***/utils/context.py:152 AirflowContextDeprecationWarning: Accessing 'prev_ds' from the template is deprecated and will be removed in a future version.
[2022-02-27 15:56:20,921] {logging_mixin.py:109} WARNING - /Users/andyhsu/tutorial-env/lib/python3.9/site-packages/***/utils/context.py:152 AirflowContextDeprecationWarning: Accessing 'prev_ds_nodash' from the template is deprecated and will be removed in a future version.
[2022-02-27 15:56:20,922] {logging_mixin.py:109} WARNING - /Users/andyhsu/tutorial-env/lib/python3.9/site-packages/***/utils/context.py:152 AirflowContextDeprecationWarning: Accessing 'prev_execution_date' from the template is deprecated and will be removed in a future version.
[2022-02-27 15:56:20,922] {logging_mixin.py:109} WARNING - /Users/andyhsu/tutorial-env/lib/python3.9/site-packages/***/utils/context.py:152 AirflowContextDeprecationWarning: Accessing 'prev_execution_date_success' from the template is deprecated and will be removed in a future version. Please use 'prev_data_interval_start_success' instead.
[2022-02-27 15:56:20,923] {logging_mixin.py:109} WARNING - /Users/andyhsu/tutorial-env/lib/python3.9/site-packages/***/utils/context.py:152 AirflowContextDeprecationWarning: Accessing 'tomorrow_ds' from the template is deprecated and will be removed in a future version.
[2022-02-27 15:56:20,923] {logging_mixin.py:109} WARNING - /Users/andyhsu/tutorial-env/lib/python3.9/site-packages/***/utils/context.py:152 AirflowContextDeprecationWarning: Accessing 'tomorrow_ds_nodash' from the template is deprecated and will be removed in a future version.
[2022-02-27 15:56:20,924] {logging_mixin.py:109} WARNING - /Users/andyhsu/tutorial-env/lib/python3.9/site-packages/***/utils/context.py:152 AirflowContextDeprecationWarning: Accessing 'yesterday_ds' from the template is deprecated and will be removed in a future version.
[2022-02-27 15:56:20,924] {logging_mixin.py:109} WARNING - /Users/andyhsu/tutorial-env/lib/python3.9/site-packages/***/utils/context.py:152 AirflowContextDeprecationWarning: Accessing 'yesterday_ds_nodash' from the template is deprecated and will be removed in a future version.
[2022-02-27 15:56:20,999] {taskinstance.py:1700} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/Users/andyhsu/tutorial-env/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 1329, in _run_raw_task
    self._execute_task_with_callbacks(context)
  File "/Users/andyhsu/tutorial-env/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 1455, in _execute_task_with_callbacks
    result = self._execute_task(context, self.task)
  File "/Users/andyhsu/tutorial-env/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 1511, in _execute_task
    result = execute_callable(context=context)
  File "/Users/andyhsu/tutorial-env/lib/python3.9/site-packages/airflow/operators/python.py", line 174, in execute
    return_value = self.execute_callable()
  File "/Users/andyhsu/tutorial-env/lib/python3.9/site-packages/airflow/operators/python.py", line 185, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/Users/andyhsu/Desktop/airflow_tutorial/airflow/dags/download_stock_price.py", line 104, in save_to_mysql_stage
    val = load_price_data(ticker)
  File "/Users/andyhsu/Desktop/airflow_tutorial/airflow/dags/download_stock_price.py", line 88, in load_price_data
    lines = reader.readlins()
AttributeError: '_io.TextIOWrapper' object has no attribute 'readlins'
[2022-02-27 15:56:21,029] {taskinstance.py:1267} INFO - Marking task as FAILED. dag_id=Download_Stock_Price, task_id=save_to_database, execution_date=20220227T075613, start_date=20220227T075620, end_date=20220227T075621
[2022-02-27 15:56:21,060] {standard_task_runner.py:89} ERROR - Failed to execute job 15 for task save_to_database
Traceback (most recent call last):
  File "/Users/andyhsu/tutorial-env/lib/python3.9/site-packages/airflow/task/task_runner/standard_task_runner.py", line 85, in _start_by_fork
    args.func(args, dag=self.dag)
  File "/Users/andyhsu/tutorial-env/lib/python3.9/site-packages/airflow/cli/cli_parser.py", line 48, in command
    return func(*args, **kwargs)
  File "/Users/andyhsu/tutorial-env/lib/python3.9/site-packages/airflow/utils/cli.py", line 92, in wrapper
    return f(*args, **kwargs)
  File "/Users/andyhsu/tutorial-env/lib/python3.9/site-packages/airflow/cli/commands/task_command.py", line 298, in task_run
    _run_task_by_selected_method(args, dag, ti)
  File "/Users/andyhsu/tutorial-env/lib/python3.9/site-packages/airflow/cli/commands/task_command.py", line 107, in _run_task_by_selected_method
    _run_raw_task(args, ti)
  File "/Users/andyhsu/tutorial-env/lib/python3.9/site-packages/airflow/cli/commands/task_command.py", line 180, in _run_raw_task
    ti._run_raw_task(
  File "/Users/andyhsu/tutorial-env/lib/python3.9/site-packages/airflow/utils/session.py", line 70, in wrapper
    return func(*args, session=session, **kwargs)
  File "/Users/andyhsu/tutorial-env/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 1329, in _run_raw_task
    self._execute_task_with_callbacks(context)
  File "/Users/andyhsu/tutorial-env/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 1455, in _execute_task_with_callbacks
    result = self._execute_task(context, self.task)
  File "/Users/andyhsu/tutorial-env/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 1511, in _execute_task
    result = execute_callable(context=context)
  File "/Users/andyhsu/tutorial-env/lib/python3.9/site-packages/airflow/operators/python.py", line 174, in execute
    return_value = self.execute_callable()
  File "/Users/andyhsu/tutorial-env/lib/python3.9/site-packages/airflow/operators/python.py", line 185, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/Users/andyhsu/Desktop/airflow_tutorial/airflow/dags/download_stock_price.py", line 104, in save_to_mysql_stage
    val = load_price_data(ticker)
  File "/Users/andyhsu/Desktop/airflow_tutorial/airflow/dags/download_stock_price.py", line 88, in load_price_data
    lines = reader.readlins()
AttributeError: '_io.TextIOWrapper' object has no attribute 'readlins'
[2022-02-27 15:56:21,100] {local_task_job.py:154} INFO - Task exited with return code 1
[2022-02-27 15:56:21,175] {local_task_job.py:264} INFO - 0 downstream tasks scheduled from follow-on schedule check
